### PR TITLE
[driver] Add driver for STTS22H temperature sensor

### PR DESCRIPTION
**Pull request by chris-durand**
Simple driver for STTS22H temperature sensor. It has been successfully tested on a custom board. I will also test the example on real hardware but not with the actual Nucleo board since the chip has a 2x2mm UDFN package.

TODO:
- [x] Test example in hardware

### CODE REVIEW

File: examples/nucleo_f103rb/stts22h/main.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	I2cMaster::connect<GpioB7::Sda, GpioB6::Scl>();
+	I2cMaster::initialize<Board::SystemClock, 100_kHz>();
+}
+
+int
+main()
+{
+	Board::initialize();
+	LedD13::setOutput();
+	initializeI2c();
```
Comment (salkinium) [inline examples/nucleo_f103rb/stts22h/main.cpp]: Why is this a separate function?

File: src/modm/driver/temperature/stts22h.hpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	MODM_FLAGS8(Status);
+	/// @endcond
+public:
+	struct modm_packed
+	Data
+	{
+		uint8_t data[2];
+
+		/// @return temperature in 1/100th °C
+		constexpr int16_t
```
Comment (salkinium) [inline src/modm/driver/temperature/stts22h.hpp:80]: WAT, constexpr I2C? Has science gone too far?2??

**Review by salkinium — COMMENTED** [review]
Please squash the docs commit into the driver commit.

**Comment by chris-durand** [comment]
> Please squash the docs commit into the driver commit.

Forgot to do it, now done.

File: examples/nucleo_f103rb/stts22h/main.cpp (hunk 1 above)
Comment (chris-durand) [reply examples/nucleo_f103rb/stts22h/main.cpp]: Copy/paste, wasn't thinking to much about it. It doesn't have to be here, but it was in some code I copied it from.

File: src/modm/driver/temperature/stts22h.hpp (hunk 2 above)
Comment (chris-durand) [reply src/modm/driver/temperature/stts22h.hpp:80]: Don't you know the [Circle language](https://www.circle-lang.org/)? It's an extension of C++17 that let's you do everything at compile time. Just throw in a `@meta` and it runs everything in your compiler, even filesystem IO, threads or I²C. Very impressive project, it has a C++17 compiler written from scratch by one person.

Comment (salkinium) [reply src/modm/driver/temperature/stts22h.hpp:80]: I'm not sure heaven exists, but I'm definitely sure hell does and everything there just runs on a single glitchy I2C bus…

**Review by salkinium — APPROVED** [review]
Any update on this? Driver looks perfectly fine to me.

**Comment by chris-durand** [comment]
The driver is working fine and tested on a custom board with an F107. I can't test the example since I don't have any spare sensors to hook up to a Nucleo board and it would also be painful with that tiny UDFN package. From my side good to merge.

**Pull request merged**